crates/gitbutler-branch-actions/src/dependencies.rs

```diff
@@ -73,6 +67,41 @@ pub fn compute_workspace_dependencies(
     })
 }
 
+/// Get the commits that need to be processed for the stack dependencies caculation.
+///
+/// Commit IDs are in the order that they are applied (parent first).
+/// Merge commits to the target branch are not included.
+fn get_commits_to_process(
+    repo: &git2::Repository,
+    stack: &Stack,
+    target_sha: &git2::Oid,
+) -> Result<Vec<git2::Oid>, anyhow::Error> {
+    let commit_ids = repo
+        .l(stack.head(), LogUntil::Commit(*target_sha), true)
+        .context("failed to list commits")?
+        .into_iter()
+        .rev()
+        .filter_map(|commit_id| {
+            let commit = repo.find_commit(commit_id).ok()?;
+            if commit.parent_count() == 1 {
+                return Some(commit_id);
+            }
+
+            let has_integrated_parent = commit.parent_ids().any(|id| {
+                let (number_commits_ahead, _) = repo.graph_ahead_behind(id, *target_sha).unwrap();
+                number_commits_ahead == 0
+            });
+
+            if has_integrated_parent {
+                None
+            } else {
+                Some(commit_id)
+            }
+        })
+        .collect_vec();
```

Comment:
Since this function is merely a filter on what could be a plain iterator, there is no need to collect it.
Instead, one can do `fn filter_iter() -> impl Iterator<Item = git2::Oid> {...}` and it will work naturally.

Probably not measurable by itself, but these things accumulate. Besides, I always find it confusing when a `Vec` is used even though there isn't a need for it.
Not a big deal as it's quite common in this codebase, I just want to create awareness.

Reply:
> Probably not measurable by itself, but these things accumulate.

Totally reasonable, will change that.
 
In general, I agree that we should improve everywhere we can, as long as the maintainability (and in turn, readability) cost is not too high.